packages/core/template/project/DeveloperGuide.md

```diff
@@ -0,0 +1,285 @@
+---
+  layout: default.md
+  title: "Developer Guide"
+  pageNav: 3
+---
+
+# ProjectEx Developer Guide
+
+<!-- * Table of Contents -->
+<page-nav-print />
+
+--------------------------------------------------------------------------------------------------------------------
+
+## **Acknowledgements**
+
+_{ list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well }_
+
+--------------------------------------------------------------------------------------------------------------------
+
+## **Setting up, getting started**
+
+Refer to the guide [_Setting up and getting started_](SettingUp.md).
+
+--------------------------------------------------------------------------------------------------------------------
+
+## **Design**
+
+<box type="tip">
+This section describes the architecture of your app, explaining how the main components work and interact with each other. Using architecture diagrams is recommended.
+</box>
+
+### Architecture
+
+The ***Architecture Diagram*** given below explains the high-level design of the ProjectEx App.
+
+<img src="images/johndoe.png" width="200px">
+<box type="info" seamless>
+
+Architecture Diagram of your project created using [`puml`](https://markbind.org/userGuide/components/imagesAndDiagrams.html#diagrams) feature.
+</box>
+
+Given below is a quick overview of main components and how they interact with each other.
+
+**Main components of the architecture**
+
+The bulk of the app's work is done by the following four components:
+
+* [**`Component 1`**](#component-1): Does something.
+* [**`Component 2`**](#component-2): Does something.
+* [**`Component 3`**](#component-3): Does something.
+
+**Interactions between architecture components**
+
+The *Sequence Diagram* below shows how the components interact with each other for a certain scenario.
+
+<img src="images/johndoe.png" width="200px">
+<box type="info" seamless>
+
+Sequence Diagram of your project created using [`puml`](https://markbind.org/userGuide/components/imagesAndDiagrams.html#diagrams) feature.
+</box>
+
+The sections below give more details of each component.
+
+<box type="tip" seamless>
+
+**Tip:** Describe in each section how each component works in detail, including its classes, methods, and how it interacts with other components.
+
+You can use a combination of Markbind's [**Diagrams**](https://markbind.org/userGuide/components/imagesAndDiagrams.html#diagrams) feature to provide a visual representation of each component, such as architecture and sequence diagrams.
+</box>
+
+### Component 1
+
+Component 1 is made up of parts e.g.`A`, `B`, `C` etc.
+
+<img src="images/johndoe.png" width="200px">
+<box type="info" seamless>
```

Comment:
Good to show both tip type boxes and info type boxes - you can consider where you can show other type of boxes ah.

One thought I have is that you might want to consider having these components in separate pages so that the scrolling is less painful. But if this is the format expected in 2103T, then lets stick with that. You might want to consider reducing the content even further though